packages/catalog-process/src/services/catalogService.ts

```diff
@@ -445,12 +445,22 @@ export function catalogServiceBuilder(
         }
       }
 
+      const updatedMode = apiEServiceModeToEServiceMode(eserviceSeed.mode);
+
+      const checkedRiskAnalysis =
+        updatedMode === eserviceMode.deliver &&
+        eservice.data.mode === eserviceMode.receive
+          ? []
+          : eservice.data.riskAnalysis;
```

Comment:
```suggestion
      const updatedMode = apiEServiceModeToEServiceMode(eserviceSeed.mode);

      const checkedRiskAnalysis =
        updatedMode !== eservice.data.mode ? [] : eservice.data.riskAnalysis;
```
@Carminepo2 shouldn't the check be like this?
In this case everytime the mode is updated the riskAnalysis is deleted as required by the task

Reply:
No, you need to keep the previous risk analysis if the mode keep being `RECEIVE`

Reply:
Yeah, this also could be a valid solution!

Reply:
Dropping a third option
```suggestion
      const updatedMode = apiEServiceModeToEServiceMode(eserviceSeed.mode);

      const checkedRiskAnalysis =
	updatedMode === eserviceMode.receive ? eservice.data.riskAnalysis : [];
```
which IMO takes the benefits of all your solutions:
- is straightforward
- manages possible future enum values
- directly highlight which is the target `mode` that we need for this logic

Reply:
Updated

Reply:
@galales Wouldn't this solution always delete the riskAnalysis on every update of an eservice with mode `receive`'?

If I am a user who wants to update the description of an eservice with mode `receive`, should I expect the riskAnalysis to be deleted?

Another thing that's not clear to me, in the task is specified that the riskAnalysis has to be deleted everytime mode is updated, but here we delete it only if `updatedMode` is `receive`, which one is the desired business logic?

Reply:
the risk analysis should be attached to an eservice only if its mode is `receive`, for all other values (for now just `deliver`) it should have an empty risk analysis.

So removing the risk analysis for a `deliver` eservice will:
- have no impacts if mode previously was already `deliver`
- remove risk analysis if mode previously was `receive`

As per task request, having now only 2 values for `mode` the two statements are equivalent.
Btw my suggestion was to keep the risk analysis if the updated is `receive`, not to delete if it is `deliver`. It's subtle but future proof in case of new `mode`s.

Does it make sense?